### ISSUE TITLE
Options to reproduce paper results

### ISSUE THREAD

**Issue by alelopes** [ISL-CV/ChiTransformer#2]
Hello,

Which training configuration is necessary to reproduce your paper results? 

I also needed to add some squeeze for the predicted depth tensor to make it work when training with some provided options using the KITTI dataset. Can you check it, please?

Thanks!

**Comment by QingSuML** [comment]
Hi,

The implementation provided here is a bit different from the version described in the paper as the tower of backbones and transformer stacks are combined into one single network to dramatically downsize the original network. Therefore, the training process is a bit different. 

Since there is no parallel monocular reprojection in the provided version in this repo, the convergence of the entire model would be a problem. To promote convergence, the monocular mode of the Chitransformer which simply reduce into a model similar to the DPT needs to be trained (self-supervised) beforehand or alongside the Chitransformer training.

I will update the repo with either a pre-trained starting point or a code for monocular mode training for people to be able to start from scratch.

For finetuning the Chitransformer on other datasets, the recommended configuration would be:
{"reprojection_loss": 1.5, 
"orthog_reg": 0.1, 
"hoyer_reg": 0.001, 
"fp_loss": 5e-05, 
"smoothness": 0.1~0.3}

if finetune on sparse ground truth, the guided loss is recommended:
{"guided_loss": 1.0, 
"supervised_loss": 0.01~0.05}

More detailed documentation on training and finetuning will be provided shortly.

Thanks.

**Comment by alelopes** [comment]
Hello @ISL-CV ,

One more question related to am curious about how exactlyctly did you implemented the dual loss for your training. I know the training process is a bit different now, and the paper's version is not available yet, but can you give me a hint on this? 

More specifically, how did you generate the outputs [("color", 'l', scale)]  results (for the left output)? 

Did you follow the same idea for the right generation but now you multiply the x-axis translation of the T matrix by -1?

Example:

Original right generation:
```            
    cam_points = self.backproject_depth[scale](depth, inputs[("inv_K", scale)])
    outputs[("sample", 'r', scale)] = self.project_3d[scale](cam_points, inputs[("K", scale)], T)
    outputs[("color", 'r', scale)] = F.grid_sample(inputs[("color", 'r', scale)],
                                                               outputs[("sample", 'r', scale)],
                                                               align_corners=True,
                                                               padding_mode="border")
```
Left image generation I suppose you did:

```           
   cam_points = self.backproject_depth[scale](depth, inputs[("inv_K", scale)])
   T_sec = T.clone()
   T_sec[:, 0, 3] = -1 * T_sec[: ,0, 3] 
   outputs[("sample", 'l', scale)] = self.project_3d[scale](cam_points, inputs[("K", scale)], T_sec)
   outputs[("color", 'l', scale)] = F.grid_sample(inputs[("color", 'l', scale)],
                                                 outputs[("sample", 'l', scale)],
                                                 align_corners=True,
                                                 padding_mode="border")
```

Thanks,
